### PR TITLE
Simplemob Girder Smashin' (For Wall Smashing Mobs Only)

### PR DESCRIPTION
**Pull request by Optimism333**
## Here's Johnny! ##
- Two boss mobs in the mothership lab vault have gotten a bit beefier, now being able to smash regular walls
- Mobs that have the SMASH_WALLS flag can now also demolish girders in their way, 25% chance per attack (demonstrated in video below)
- The girder leaves two sheets of its material behind after being destroyed, so it can potentially be rebuilt.

https://user-images.githubusercontent.com/81007356/170432267-72e72096-26d9-4069-bb5c-1d6aa15c4fdd.mp4

## Why you do this ##
- I thought it was weird that mobs capable of breaking walls couldn't break girders. So I decided to see if I could fix it. It turns out, fixing it was easier than I thought! Hopefully the code's not too terrible.
- Girder smashing could probably be a bit better refined, open to suggestions.

:cl:
 * rscadd: Simplemobs that can smash walls can now also smash girders (25% chance per attack)
 * rscadd: Two of the boss mobs in the ayy vault can now smash basic walls

### CODE REVIEW

**Comment by tuypo1** [comment]
I'm fine with R girders being as easy to smash as regular walls

**Comment by BomberBro** [comment]
Should take more time instead of the mob sending the girder to another dimension instantly by staring at it

**Comment by west3436** [comment]
unatomic

**Comment by Optimism333** [comment]
> Should take more time instead of the mob sending the girder to another dimension instantly by staring at it

Fair. I'll see if I can try to change how this works

**Comment by Optimism333** [comment]
> unatomic

The pr is now more atomic

**Comment by Servalfang** [comment]
it really didnt need to be super atomic its an obscure change and a not obscure change, the problem would be more multiple very clear general-round-affecting change

**Comment by Optimism333** [comment]
The way the girder smashing works has been slightly changed. The mob has a 25% chance to bust the girder per attack, rather than having it be guaranteed. I put Lawyer Larry through the ringer a few times to see how this looks, and I do think it's a bit of an improvement. Still could probably be better.

https://user-images.githubusercontent.com/81007356/170121385-0948049d-6657-46d4-94f4-c9c02a9c6c0c.mp4


https://user-images.githubusercontent.com/81007356/170121410-884690dc-ae59-427c-921c-6508c4ec532f.mp4

**Comment by pedr69** [comment]
>destroyed girders don't drop metal

**Comment by Optimism333** [comment]
> > destroyed girders don't drop metal

Will have a look at this and see what I can do.

**Comment by Hinaichigo** [comment]
Wall smashing has 100% chance to work on the first hit and since girders are structural elements of walls it does seem a little counterintuitive to have them harder to destroy at 25%. But the 25% is tending towards a notion of wall health which I like. Overall it's a nice change either way.

What if smashing a wall would produce a girder that would then have to be smashed to fully break through?

**Comment by Optimism333** [comment]
> Wall smashing has 100% chance to work on the first hit and since girders are structural elements of walls it does seem a little counterintuitive to have them harder to destroy at 25%. But the 25% is tending towards a notion of wall health which I like. Overall it's a nice change either way.
> 
> What if smashing a wall would produce a girder that would then have to be smashed to fully break through?

Basically agree with everything you said, and would very much like if wall smashing could be moved to a more gradual breakthrough rather than instantly demolishing the wall. It'd be neat if in general mobs took a little longer to break through obstacles but could break through more of them. There's a fairly large list of things that they don't know how to deal with currently if they get blocked by them (filing cabinets, floodlights, machine frames, computers) just to name a few.

However, I'm going to leave this as a relatively small change for now. Maybe I'll revisit this part of mob code at some point.

**Comment by Optimism333** [comment]
https://user-images.githubusercontent.com/81007356/170431621-6c55366b-a13d-4d3c-8bdf-f7c1b0213231.mp4

Girders now also drop 2 of their material when destroyed. (ie: basic girder will drop two metal)

File: code/game/objects/structures/girders.dm [hunk 1]
```diff
@@ -6,6 +6,20 @@
...
 	pass_flags_self = PASSGIRDER
+
+/obj/structure/girder/attack_animal(var/mob/living/simple_animal/M)
+	M.delayNextAttack(8)
+	if(M.environment_smash_flags & SMASH_WALLS)
+		if(prob(25)) // Not the best solution, but this should allow for better feedback so the player realizes the mob is trying to break through and has time to retreat
+			playsound(src, 'sound/weapons/heavysmash.ogg', 75, 1)
+			M.visible_message("<span class='danger'>[M] smashes through \the [src].</span>", \
+			"<span class='attack'>You smash through \the [src].</span>")
+			new material(get_turf(src), 2)
```
Comment (DamianX) [inline code/game/objects/structures/girders.dm]:
```suggestion
			drop_stack(material, get_turf(src), 2)
```

**Pull request merged**